File: 2020/Python/day1.py
from aoc import *

def solve(inp):
    nums = set(intcolumn(inp)) #helper function to load a column integer vector from string
    for a in nums:
        for b in nums - {a}:
            if a + b == 2020:
                part1 = a * b
            c = 2020 - a - b
            if c in nums:
                part2 = a * b * c
    return part1, part2


run(solve)

File: 2020/Nim/day1.nim
import aoc

func solve*(input:string): (int, int) =
    var numbers: set[int16]
    for n in input.parseInts:
        numbers.incl(n.int16)
    var part1, part2: int
    for a in numbers:
        for b in numbers - {a}:
            if a + b == 2020:
                part1 = a.int * b.int
            let c = 2020 - a - b
            if c in numbers:
                part2 = a.int * b.int * c.int
    return (part1, part2)